packages/framework-provider-kubernetes/src/services/read-model-registry.ts

```diff
@@ -23,27 +23,44 @@ export class ReadModelRegistry {
     filters: Filters
   ): Promise<Array<ReadModelInterface>> {
     const l = (msg: string): void => logger.debug('readModelRegistry#search: ' + msg)
-    const readModelId = filters.id.eq
-    l(`Got id ${readModelId ?? 'UNDEFINED'}`)
-    if (!readModelId) throw new Error('Only searching by ID is supported')
-    const url = `${this.url}/v1.0/state/statestore/${this.readModelKey(readModelName, readModelId)}`
-    l(`Performing a fetch to ${url}`)
-    const response = await fetch(url)
-    if (!response.ok) {
-      l(`Error on performing GET from ${url}`)
-    }
-    l(`Got result ${JSON.stringify(response)}`)
-    try {
-      const body = await response.json()
-      l(`Got JSON ${JSON.stringify(body)}`)
-      // TODO: Remove unnecessary id setting from k8s read model registry
-      if (body) {
-        body.id = readModelId
+    if (filters.id) {
+      // TODO: Allow complex filters in K8s provider. Currently we only allow filter by Id or get all entries
+      const readModelId = filters.id.eq
+      l(`Got id ${readModelId ?? 'UNDEFINED'}`)
+      if (!readModelId) throw new Error('Only searching by ID is supported')
+      const url = `${this.url}/v1.0/state/statestore/${this.readModelKey(readModelName, readModelId)}`
+      l(`Performing a fetch to ${url}`)
+      const response = await fetch(url)
+      if (!response.ok) {
+        l(`Error on performing GET from ${url}`)
+      }
+      l(`Got result ${JSON.stringify(response)}`)
+      try {
+        const body = await response.json()
+        l(`Got JSON ${JSON.stringify(body)}`)
+        // TODO: Remove unnecessary id setting from k8s read model registry
+        if (body) {
+          body.id = readModelId
+        }
+        return [body.value]
+      } catch (err) {
+        l(`Error ${JSON.stringify(err)}`)
+        return []
       }
-      return [body.value]
-    } catch (err) {
-      l(`Error ${JSON.stringify(err)}`)
-      return []
+    } else {
+      const keys = await this.redis.keys(`rm_${readModelName}_*`, logger)
+      l(`Obtainer following keys for query: ${keys}`)
+      const results: ReadModelInterface[] = []
```

Comment:
I'm not really happy with this implementation. 

I was having issues while dealing with types and the map function because the `value` inside the map can be undefined and that forces the return type to be `ReadModelEvenloper | undefined []` and this type is different from the return type 🤷🏼‍♂️ . 

Using the `results` array, I avoid the type being `undefined` but I think that there is a more elegant way to solve this.

Reply:
Agree, but we can improve implementations once we have something working